mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -407,6 +449,103 @@ private[classification] class MultiClassSummarizer extends Serializable {
   }
 }
 
+@Experimental
+/**
+ * :: Experimental ::
+ * Logistic regression training results.
+ * @param predictions dataframe outputted by the model's `transform` method.
+ * @param probabilityCol field in "predictions" which gives the calibrated probability of
+ *                       each sample as a vector.
+ * @param labelCol field in "predictions" which gives the true label of each sample.
+ * @param objectiveHistory objective function (scaled loss + regularization) at each iteration.
+ */
+class LogisticRegressionTrainingSummary private[classification] (
+    predictions: DataFrame,
+    probabilityCol: String,
+    labelCol: String,
+    val objectiveHistory: Array[Double])
+  extends LogisticRegressionSummary(predictions, probabilityCol, labelCol) {
+
+  /** Number of training iterations until termination */
+  val totalIterations = objectiveHistory.length
+
+}
+
+@Experimental
+/**
+ * :: Experimental ::
+ * Logistic regression results for a given model.
+ * @param predictions dataframe outputted by the model's `transform` method.
+ * @param probabilityCol field in "predictions" which gives the calibrated probability of
+ *                       each sample.
+ * @param labelCol field in "predictions" which gives the true label of each sample.
+ */
+class LogisticRegressionSummary private[classification] (
+  @transient val predictions: DataFrame,
+  val probabilityCol: String,
+  val labelCol: String) extends Serializable {
+
+  private val sqlContext = predictions.sqlContext
+  import sqlContext.implicits._
+
+  /** Returns a BinaryClassificationMetrics object.
+  */
+  // TODO: Allow the user to vary the number of bins using a setBins method in
+  // BinaryClassificationMetrics. For now the default is set to 100.
+  @transient private val metrics = new BinaryClassificationMetrics(
+    predictions.select(probabilityCol, labelCol).map {
+      case Row(score: Vector, label: Double) => (score(1), label)
+    }, 100
+  )
+
+  /**
+   * Returns the receiver operating characteristic (ROC) curve,
+   * which is an Dataframe having two fields (false positive rate, true positive rate)
+   * with (0.0, 0.0) prepended and (1.0, 1.0) appended to it.
+   * Every possible probability obtained in transforming the dataset are used
+   * as thresholds used in calculating the FPR and TPR.
+   */
+  def roc(): DataFrame = metrics.roc().toDF("FalsePositiveRate", "TruePositiveRate")
+
+  /**
+   * Computes the area under the receiver operating characteristic (ROC) curve.
+   */
+  def areaUnderROC(): Double = metrics.areaUnderROC()
+
+  /**
+   * Returns the precision-recall curve, which is an Dataframe containing
+   * two fields (recall, precision) NOT (precision, recall), with (0.0, 1.0) prepended to it.
+   * Every possible probability obtained in transforming the dataset are used
```

Comment:
I would remove this sentence, and other copies of it below.  I don't think it's true since metrics uses binning, and I think modifying it would require more explanation.